lambda_function.py

```diff
@@ -1,218 +1,114 @@
-        
-    for service_class in service_class_info:
-        queued_metric = {}
-        queued_metric['MetricName'] = 'ServiceClass%s-Queued' % service_class[0]
-        queued_metric['Dimensions'] = [{'Name': 'ClusterIdentifier', 'Value': cluster}]
-        queued_metric['Timestamp'] = poll_ts
-        queued_metric['Value'] = service_class[1]
-        metrics.append(queued_metric.copy())
-
-        executing_metric = {}
-        executing_metric['MetricName'] = 'ServiceClass%s-Executing' % service_class[0]
-        executing_metric['Dimensions'] = [{'Name': 'ClusterIdentifier', 'Value': cluster}]
-        executing_metric['Timestamp'] = poll_ts
-        executing_metric['Value'] = service_class[2]
-        metrics.append(executing_metric.copy())
-
-    return metrics
-
-
-def gather_table_stats(cursor, cluster):
-    run_command(cursor, "select /* Lambda CloudWatch Exporter */ \"schema\" || '.' || \"table\" as table, encoded, max_varchar, unsorted, stats_off, tbl_rows, skew_sortkey1, skew_rows from svv_table_info")
-    tables_not_compressed = 0
-    max_skew_ratio = 0
-    total_skew_ratio = 0
-    number_tables_skew = 0
-    number_tables = 0
-    max_skew_sort_ratio = 0
-    total_skew_sort_ratio = 0
-    number_tables_skew_sort = 0
-    number_tables_statsoff = 0
-    max_varchar_size = 0
-    max_unsorted_pct = 0
-    total_rows = 0
-    
-    result = cursor.fetchall()
-    
-    for table in result:
-        table_name, encoded, max_varchar, unsorted, stats_off, tbl_rows, skew_sortkey1, skew_rows = table
-        number_tables += 1
-        if encoded == 'N':
-            tables_not_compressed += 1
-        if skew_rows != None:
-            if skew_rows > max_skew_ratio:
-                max_skew_ratio = skew_rows
-            total_skew_ratio += skew_rows
-            number_tables_skew += 1
-        if skew_sortkey1 != None:
-            if skew_sortkey1 > max_skew_sort_ratio:
-                max_skew_sort_ratio = skew_sortkey1
-            total_skew_sort_ratio += skew_sortkey1
-            number_tables_skew_sort += 1
-        if stats_off != None and stats_off > 5:
-            number_tables_statsoff += 1
-        if max_varchar != None and max_varchar > max_varchar_size:
-            max_varchar_size = max_varchar
-        if unsorted != None and unsorted > max_unsorted_pct:
-            max_unsorted_pct = unsorted
-        if tbl_rows != None:
-            total_rows += tbl_rows
-
-    if number_tables_skew > 0:
-        avg_skew_ratio = total_skew_ratio / number_tables_skew
-    else:
-        avg_skew_ratio = 0
+            avg_skew_sort_ratio = 0
 
-    if number_tables_skew_sort > 0:
-        avg_skew_sort_ratio = total_skew_sort_ratio / number_tables_skew_sort
-    else:
-        avg_skew_sort_ratio = 0
-    
-    # build up the metrics to put in cloudwatch
-    return [
+        # build up the metrics to put in cloudwatch
+        return [
             {
                 'MetricName': 'TablesNotCompressed',
```

Comment:
These metrics dicts are so similar...does it make sense to abstract them into a NamedTuple or something like that?

Reply:
I did not want to mess with it too much, but yeah, that should be abstracted, I suggest we do it if/when they accept the PR, or if they don't we do it in our version.

Reply:
cool. didn't realize we were gonna (try to) contribute back, but that's great